src/resolve-config.js

```diff
@@ -20,23 +22,47 @@ function getLoadFunction(opts) {
   return getExplorerMemoized(opts).load;
 }
 
-function resolveConfig(filePath, opts) {
+function _resolveConfig(filePath, opts, sync) {
   opts = Object.assign({ useCache: true }, opts);
-  const load = getLoadFunction({ cache: !!opts.useCache, sync: false });
-  return load(filePath, opts.config).then(result => {
-    return !result ? null : mergeOverrides(result, filePath);
-  });
+  const loadOpts = { cache: !!opts.useCache, sync: !!sync };
+  const load = getLoadFunction(loadOpts);
+  const loadEditorConfig = resolveEditorConfig.getLoadFunction(loadOpts);
+  const arr = [load, loadEditorConfig].map(l => l(filePath, opts.config));
+
+  const unwrapAndMerge = arr => {
+    const result = arr[0];
+    const editorConfigured = arr[1];
+    return mergeEditorConfig(filePath, result, editorConfigured);
+  };
+
+  if (loadOpts.sync) {
+    return unwrapAndMerge(arr);
+  }
+
+  return Promise.all(arr).then(unwrapAndMerge);
 }
 
-resolveConfig.sync = (filePath, opts) => {
-  opts = Object.assign({ useCache: true }, opts);
-  const load = getLoadFunction({ cache: !!opts.useCache, sync: true });
-  const result = load(filePath, opts.config);
-  return !result ? null : mergeOverrides(result, filePath);
-};
+const resolveConfig = (filePath, opts) => _resolveConfig(filePath, opts, false);
+
+resolveConfig.sync = (filePath, opts) => _resolveConfig(filePath, opts, true);
+
+function mergeEditorConfig(filePath, result, editorConfigured) {
+  if (!filePath) {
+    return null;
```

Comment:
Unrelated: it'd be better to add a comment to describe why we return `null` here since our docs now says `filePath` is a required parameter.

Reply:
It turns out we could get rid of this altogether. Fixed in https://github.com/prettier/prettier/pull/2760/commits/49a19fe8ebf7c5b30d99d0023147c28257939df3